cmd/kubeadm/app/cmd/join.go

```diff
@@ -419,6 +421,20 @@ func (j *joinData) Run() error {
 		fmt.Printf("[join] Running pre-flight checks before initializing the new control plane instance\n")
 		preflight.RunInitMasterChecks(utilsexec.New(), initCfg, j.ignorePreflightErrors)
 
+		fmt.Printf("[join] Pulling control-plane images\n")
+		externalcfg := &kubeadmapiv1beta1.InitConfiguration{}
+		kubeadmscheme.Scheme.Default(externalcfg)
+		var cfgPath string
+		internalcfg, err := configutil.ConfigFileAndDefaultsToInternalConfig(cfgPath, externalcfg)
+		kubeadmutil.CheckErr(err)
+		containerRuntime, err := utilruntime.NewContainerRuntime(utilsexec.New(), internalcfg.GetCRISocket())
+		kubeadmutil.CheckErr(err)
```

Comment:
please note that `CheckErr` causes kubeadm to exit on error.
check for errors using `if err != nil....`

Reply:
yep :man_facepalming:  Thx agree!